src/protocol/vm/adapter_contract.rs

```diff
@@ -214,7 +215,7 @@ where
                                 "Adapter price calculation failed: Denominator is zero".to_string(),
                             ))
                         } else {
-                            Ok((numerator.as_u128() as f64) / (denominator.as_u128() as f64))
+                            Ok(u256_to_f64(numerator) / u256_to_f64(denominator))
```

Comment:
It converts to uint
`pub type Uint = ethereum_types::U256;`

I think an `f64` would be safer before division, otherwise you're trying to divide two uint256s?
See for example how it's done in `sqrt_price_q96_to_f64`

Reply:
ok ok! I agree! Thank you!